src/Hosting.CommandLine/HostBuilderExtensions.cs

```diff
@@ -0,0 +1,69 @@
+using McMaster.Extensions.CommandLineUtils;
+using McMaster.Extensions.Hosting.CommandLine.Internal;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using System.Threading.Tasks;
+
+namespace McMaster.Extensions.Hosting.CommandLine
+{
+    /// <summary>
+    /// Extension methods for <see cref="IHostBuilder"/> support.
+    /// </summary>
+    /// <seealso href="https://github.com/natemcmaster/CommandLineUtils/issues/134">host support</seealso>
+    public static class HostBuilderExtensions
+    {
+        /// <summary>
+        /// Runs <code>T</code> as a <see cref="CommandLineApplication"/> with the
+        /// supplied <code>args</code>.  This method should be the primary approach 
+        /// taken for command line applications.
+        /// </summary>
+        /// <typeparam name="T">The command line application implementation</typeparam>
+        /// <param name="hostBuilder">This instance</param>
+        /// <param name="args">The command line arguments</param>
+        /// <returns>A task whose result is the exit code of the application</returns>
+        public static async Task<int> RunCliAsync<T>(this IHostBuilder hostBuilder, string[] args) where T : class
```

Comment:
What do you think about naming this `RunCommandLineAppAsync<T>`?

Reply:
I think its a [good idea](https://github.com/natemcmaster/CommandLineUtils/pull/167/commits/7ceb160b0784ad1c90040d98ab9f5faa48625aa1)